contributingGuides/PERFORMANCE.md

```diff
@@ -1,52 +1,264 @@
+- **Mobile:** Built-in - press `j` in Metro terminal to open React DevTools
+
+#### Steps to Profile:
+
+1. **Configure Settings:**
+   - Open React DevTools and go to the Profiler tab
+   - Click the settings icon
+   - Enable "Record why each component rendered while profiling" for detailed insights
+   - Optionally set "Hide commits below X ms" to focus on slow renders
+
+2. **Capture React Performance:**
+   - Click the record button (circle icon) to start profiling
+   - Perform the actions you want to analyze
+   - Click stop to end the recording
+
+3. **Analyze Results:**
+   - Review the flame graph showing component render times
+   - Use the timeline to identify slow commits
+   - Click on components to see why they rendered
+   - Look for components with high render times or frequent re-renders
+
+#### Key Metrics to Focus On:
+- **Commit duration:** Total time for a render cycle
+- **Component render time:** Individual component performance
+- **Render reasons:** Why components re-rendered (props change, state change, parent render)
+
+#### Compare the results
+
+When working with React Profiler, you can compare the results using [this tool](https://kacper-mikolajczak.github.io/rcc/). It will tell you the time and commit difference between two or more profiles.
 
 **Suggested:** [Deep Dive with the React DevTools creator](https://www.youtube.com/watch?v=nySib7ipZdk)
 
-### Why Did You Render?
-- Why Did You Render (WDYR) sends console notifications about potentially avoidable component re-renders.
-- It can also help to simply track when and why a certain component re-renders.
-- To enable it, set `USE_WDYR=true` in your `.env` file.
-- You can add or exclude tracked components by their `displayName` in `wdyr.js`.
-- Open the browser console to see WDYR notifications.
+### Tracking Re-renders
+
+Understanding when and why components re-render is crucial for identifying performance bottlenecks. Several tools can help visualize and track component re-renders in real-time.
+
+#### React DevTools Highlight Updates
+
+React DevTools includes a built-in feature to visually highlight components when they re-render:
+
+- Open React DevTools
+- Go to "Settings" (gear icon)
+- Under "General" tab, enable "Highlight updates when components render"
+- Components will be highlighted with colored borders when they update
+
+#### React Scan
+
+[React Scan](https://github.com/aidenybai/react-scan) automatically detects and highlights performance issues in React applications without requiring installation:
+
+**Usage:**
+```bash
+npx react-scan https://localhost:8082
+```
+
+You need to have the web server running. The app will open in a separate browser window.
+
+#### Why Did You Render?
+
+Why Did You Render (WDYR) sends console notifications about potentially avoidable component re-renders:
+
+- To enable it, set `USE_WDYR=true` in your `.env` file
+- You can add or exclude tracked components by their `displayName` in `wdyr.js`
```

Comment:
Does this work for you?

Reply:
Haven't used it in a while, but it worked for me some time ago (at least on web)

Reply:
Can you please confirm this before we add it to the instructions 😄